lib/puppet/provider/service/launchd.rb

```diff
@@ -1,4 +1,4 @@
-require 'facter/util/plist'
+require 'puppet/util/plist'
```

Comment:
This adds a hard dependency on `cfpropertylist` (via `puppet/util/plist`), which will prevent puppet code from being loaded on any system that doesn't have the `cfpropertylist` gem, e.g. on the master during compilation. The typical way to create a soft dependency from a provider is to create a `cfpropertylist` feature in `lib/puppet/features`:

```
require 'puppet/util/feature'
Puppet.features.add(:cfpropertylist, :libs => ["cfpropertylist"])
```

And then confine the provider to that feature:

```
confine :feature => :cfpropertylist
```